src/effects/native/bessel4lvmixeqeffect.cpp

```diff
@@ -187,25 +184,66 @@ void Bessel4LVMixEQEffect::processChannel(const ChannelHandle& handle,
     //    pOutput[i + 1] = pState->m_pBandBuf[i];
     //}
 
-    if (fLow != pState->old_low ||
-            fMid != pState->old_mid ||
-            fHigh != pState->old_high) {
-        SampleUtil::copy3WithRampingGain(pOutput,
-                pState->m_pLowBuf, pState->old_low, fLow,
-                pState->m_pBandBuf, pState->old_mid, fMid,
-                pState->m_pHighBuf, pState->old_high, fHigh,
-                numSamples);
-    } else {
+    if (fLow == pState->old_low &&
+            fMid == pState->old_mid &&
+            fHigh == pState->old_high) {
         SampleUtil::copy3WithGain(pOutput,
                 pState->m_pLowBuf, fLow,
                 pState->m_pBandBuf, fMid,
                 pState->m_pHighBuf, fHigh,
                 numSamples);
-    }
+    } else {
+        int copySamples = 0; 
+        int rampingSamples = numSamples; 
+        if ((fLow && !pState->old_low) ||
+                (fMid && !pState->old_mid) ||
+                (fHigh && !pState->old_high)) {
+            // we have just switched at least one filter on
+            // Hold of ramping for the group delay
+            if (pState->m_rampHoldOff == kRampDone) {
+                // multiply the group delay * 2 to ensure that the filter is settled
+                // it is actually at a factor of 1,8 at default setting
+                pState->m_rampHoldOff = pState->m_groupDelay * 2;
+                // ensure that we have at least 128 samples (our smallest buffer)
+                // for ramping 
+                int rampingSamples = numSamples - (pState->m_rampHoldOff % numSamples);
+                if (rampingSamples < 128) {
```

Comment:
There is nothing particular risky here. The code will work with any buffer size. 

Using less then 1 ms buffer-size will fail our de-click clock code all over Mixxx, since a 1 ms ramp produces a noise with a main lobe at  250 Hz, which starts to be hear-able.  
So I do not think that it is required to fix it only here.